src/gsy_e/gsy_e_core/simulation.py

```diff
@@ -608,9 +607,9 @@ def run_simulation(setup_module_name="", simulation_config=None, simulation_even
                 redis_job_id=saved_sim_state["general"]["simulation_id"],
                 **kwargs
             )
-
-    except GSyException as ex:
-        raise click.BadOptionUsage(ex.args[0])
+    except SimulationException as ex:
+        log.error(ex)
+        return
```

Comment:
I am not sure about this. AFAIS, this will break our error handling in the rq_job_handler:
https://github.com/gridsingularity/gsy-e/blob/0854c61564645deaaf12beb954a4e42c23a735f7/src/gsy_e/gsy_e_core/rq_job_handler.py#L126
I think we need to re-raise the exception here.

Reply:
Actually it will not, because I explicitly caught SimulationException, which is only raised in this context by ModuleNotFoundError or ImportError. All other exceptions will reach the rq_job_handler , only without wrapping them up in click.BadOptionUsage errors. And this is actually one of the fixes, because click does not make sense to use in non-cli mode, it should only be used when someone runs the software through cli. Therefore its exceptions should not be used here, and we can use the exceptions that the code itself produces.

Reply:
I understand now. Thanks a lot!